app/scripts/modules/core/src/presentation/forms/fields/ExpressionRegexField.tsx

```diff
@@ -0,0 +1,119 @@
+    ExpressionRegexField,
+  );
+  static defaultProps: Partial<IExpressionFieldProps> = { markdown: false, FieldLayout: BasicLayout };
+  public state: IExpressionRegexFieldState = {
+    spelPreview: null,
+    spelError: null,
+  };
+
+  private renderRegexFields(props: IExpressionRegexFieldProps) {
+    const { RegexHelp, regex, replace, onRegexChanged, onReplaceChanged } = props;
+
+    return (
+      <CollapsibleSection
+        heading={({ chevron }) => (
+          <span>
+            {chevron} Regex {RegexHelp}
+          </span>
+        )}
+        outerDivClassName=""
+        toggleClassName="clickable"
+        defaultExpanded={!!regex}
+      >
+        <div className="flex-container-h baseline">
+          <code>s/</code>
+          <input
+            type="text"
+            value={regex || ''}
+            onChange={evt => onRegexChanged(evt.target.value)}
+            className="flex-grow"
+          />
+          <code>/</code>
+          <input
+            type="text"
+            value={replace || ''}
+            onChange={evt => onReplaceChanged(evt.target.value)}
+            className="flex-grow"
+          />
+          <code>/g</code>
+        </div>
+      </CollapsibleSection>
+    );
+  }
+
+  public render() {
+    const { spelError, spelPreview } = this.state;
+    const { value, onChange, placeholder, markdown, context, FieldLayout, regex, replace } = this.props;
+
+    let finalSpelPreview = spelPreview;
+    let regexError: string = null;
+
+    if (spelPreview && regex) {
+      try {
+        console.log('spelPreview', spelPreview);
+        finalSpelPreview = spelPreview.replace(new RegExp(regex, 'g'), replace);
+      } catch (err) {
+        regexError = err.toString();
+      }
+    }
+
+    const expressionInput = (
+      <ExpressionInput
+        value={value}
+        onChange={evt => onChange(evt.target.value)}
+        onExpressionChange={exprChange => this.setState(exprChange)}
+        context={context}
+        placeholder={placeholder}
+      />
+    );
+
+    const regexFields = this.renderRegexFields(this.props);
+
+    const input = (
+      <div className="flex-container-v flex-grow">
+        {expressionInput}
+        {regexFields}
+      </div>
+    );
+
+    const { label, help, actions, error: externalError, warning: externalWarning } = this.props;
+
+    const error =
+      (externalError && <ValidationError message={externalError} />) ||
+      (context && spelError && <ExpressionError spelError={spelError} />) ||
+      (context && regexError && <ValidationError message={regexError} />);
+    const warning = externalWarning && <ValidationWarning message={externalWarning} />;
+    const preview = finalSpelPreview && <ExpressionPreview spelPreview={finalSpelPreview} markdown={markdown} />;
+
+    const layoutProps: IFieldLayoutProps = { label, input, help, actions, error, warning, preview };
+
+    return <FieldLayout {...layoutProps} />;
```

Comment:
A `Field` component renders itself using the `FieldLayout` prop and all the `IFieldLayoutProps` either passed in from above, or computed within the Field itself (see `error` for an example)